contracts-test/TestSimpleOracle.sol

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "../contracts/modules/common/SimpleOracle.sol";
+
+contract TestSimpleOracle is SimpleOracle {
+
+    bytes32 internal creationCode;
+
+    constructor(address _uniswapRouter) SimpleOracle(_uniswapRouter) public {
+        address uniswapV2Factory = IUniswapV2Router01(_uniswapRouter).factory();
```

Comment:
I think this line is unnecessary as you already inherit `uniswapV2Factory` from `SimpleOracle`